common/src/java/org/apache/hadoop/hive/common/type/TimestampTZUtil.java

```diff
@@ -125,11 +125,6 @@ public static TimestampTZ parseOrNull(String s, ZoneId defaultTimeZone) {
     }
   }
 
-  // Converts Date to TimestampTZ.
-  public static TimestampTZ convert(Date date, ZoneId defaultTimeZone) {
-    return parse(date.toString(), defaultTimeZone);
-  }
-
   // Converts Timestamp to TimestampTZ.
   public static TimestampTZ convert(Timestamp ts, ZoneId defaultTimeZone) {
```

Comment:
Nice catch, this is indeed an unnecessarily slow solution! This other convert function right next to it has the same issue, please fix that too in the same patch!